common/locks/priority_semaphore_impl.go

```diff
@@ -70,7 +70,6 @@ func (s *PrioritySemaphoreImpl) Acquire(ctx context.Context, priority Priority,
 			if priority == PriorityHigh {
 				s.highWaiting++
 				s.highCV.Wait()
-				s.highWaiting--
```

Comment:
wait, this seems wrong now: on a spurious wakeup it should leave highWaiting where it was before but now it'll be too high

Reply:
It is wrong now. In a spurious wake up we'll keep incrementing

Reply:
Actually this had to be an accident: there's no decrement of this any more

Reply:
Oh! I just removed it from here and put it at line 124. That made this wrong. I had to put it at line 124 to reduce highWaiting when signaling a highwaiting thread. Otherwise Test_AllThreadsAreWokenUp() will fail. Let me fix this.